strings.md added in the kotlin folder (#1472)
Codecademy/docs: content/kotlin/concepts/strings/strings.md

---
Title: 'Strings'
Description: 'A string in Kotlin is an object that holds a sequence of characters contained within a pair of double quotes (").'
Subjects:
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Data Types'
  - 'Characters'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

In Kotlin, a `string` is an immutable object that represents a sequence of characters.

## Syntax

```pseudo
val stringName: String = "string value"
```

## Example

> You can also use `string` templates to include dynamic values in your strings. To do this, you enclose the dynamic value in curly braces and precede it with a dollar sign, like this:

Reviewer: You should separate string templates into its own section after the example. Call it "## String Templates" and provide examples that produce some sort of output.

Author: @SSwiniarski hmmm alright so you want me to keep this string template section separate from the example section? like I can remove all this and make a diff header as string templates like you said

Author: alright check now

```kotlin
val dynamicValue = 4

> val stringName: String = "The value of dynamicValue is $dynamicValue"

Reviewer: You're using an example without the curly braces and this will lead to confusion. You should explain when the curly braces are optional and show examples with and without.

Author: wait I will add an example with curly braces then

---
Title: 'Strings'
Description: 'Strings are immutable objects that are sequences of characters contained within a pair of double quotes.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Data Types'
  - 'Characters'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> **Strings** are immutable objects that represents a sequence of characters contained within double quotes (`""`).

Reviewer 2: ```suggestion
**Strings** are immutable objects that represent a sequence of characters contained within double quotes (`""`).
```

## Syntax

```pseudo
val stringName: String = "string value"
```

> In Kotlin, string values are always defined as instances of the `String` class.

Reviewer 2: ```suggestion
In Kotlin, string values are always defined as instances of the `String` class.

### String Templates

Templates can also be used to dynamically produce string values. Dynamic values don't have to start as strings, and can either be directly referenced with a dollar sign (`$`) or evaluated as a string with additional curly braces (`{}`).

```

Reviewer 2: After a second look, I'd suggest placing the information about string templates directly below the end of the "Syntax" section and changing to a lv. 3 header.

## Example

> Here is an example of how you might use strings in Kotlin:

Reviewer 2: ```suggestion
The following example showcases the various ways strings are used in Kotlin:
```

```kotlin

> fun main(args: Array<String>) {
>   val greeting: String = "Hello, World!"
>   println(greeting)
> }

Reviewer 2: ```suggestion
fun main(args: Array<String>) {
  // String literals
  val greeting: String = "Hello, World!"
  println(greeting)
  
  // String templates
  val dynamicValue = 4
  val stringOne: String = "The value of dynamicValue is $dynamicValue"
  System.out.println(stringOne)
  
  val a = dynamicValue
  val b = 5
  val stringTwo: String = "The sum of a and b is ${a + b}"
  System.out.println(stringTwo)
}
```

```

The output for the above code will be:

```shell

> Hello, World!

Reviewer 2: ```suggestion
Hello, World!
The value of dynamicValue is 4
The sum of a and b is 9
```

```

> ## String Templates
> 
> Templates can also be used to dynamically produce string values. Dynamic values don't have to start as strings, and can either be directly referenced with a dollar sign (`$`) or evaluated as a string with additional curly braces (`{}`). The following showcases both techniques:
> 
> ```kotlin
> val dynamicValue = 4
> val stringOne: String = "The value of dynamicValue is $dynamicValue"
> System.out.println(stringOne)
> 
> val a = 4
> val b = 5
> val stringTwo: String = "The sum of a and b is ${a + b}"
> System.out.println(stringTwo)
> ```
> 
> This will print the following output:
> 
> ```shell
> The value of dynamicValue is 4
> The sum of a and b is 9
> ```

Reviewer 2: ```suggestion
```

---
Title: 'Strings'
Description: 'A string in Kotlin is an object that holds a sequence of characters contained within a pair of double quotes (").'
Subjects:
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Data Types'
  - 'Characters'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

In Kotlin, a `string` is an immutable object that represents a sequence of characters.

## Syntax

```pseudo
val stringName: String = "string value"
```

## Example

Here is an example of how you might use strings in Kotlin:

```kotlin
fun main(args: Array<String>) {
    val greeting: String = "Hello, World!"
    println(greeting)
}
```

The output for the above code will be:

```
Hello, World!
```

## String Templates

> We can also use `string` templates to include dynamic values in your strings. To do this, you enclose the dynamic value in curly braces and precede it with a dollar sign, like this:

Reviewer 2: ```suggestion
Templates can also be used to dynamically produce string values. Dynamic values don't have to start as strings, and can either be directly referenced with a dollar sign (`$`) or evaluated as a string with additional curly braces (`{}`). The following showcases both techniques:
```

```kotlin
val dynamicValue = 4

> val stringName: String = "The value of dynamicValue is $dynamicValue"

Reviewer 2: ```suggestion
val stringOne: String = "The value of dynamicValue is $dynamicValue"
System.out.println(stringOne)

val a = 4
val b = 5
val stringTwo: String = "The sum of a and b is ${a + b}"
System.out.println(stringTwo)
```

Reviewer 2: I would suggest combining both template versions in a single \```kotlin code block and adding another `printlin()` statement, as shown above.

```

> 
> Below we will see an example with use of curly braces `{}`:
> 
> ```java
> val a = 4
> val b = 5
> val stringName: String = "The sum of a and b is ${a + b}"
> System.out.println(stringName)
> ```

Reviewer 2: ```suggestion
```

Since Kotlin is a supported language for code blocks, the \```java signature is not necessary.

> The output for the above code will be:

Reviewer 2: ```suggestion
This will print the following output:
```

> ```

Reviewer 2: ```suggestion
```shell
```

> The sum of a and b is 9

Reviewer 2: ```suggestion
The value of dynamicValue is 4
The sum of a and b is 9
```
